src/sidebar/services/test/threads-test.js

```diff
@@ -0,0 +1,88 @@
+import threadsService from '../threads';
+
+const NESTED_THREADS = {
+  id: 'top',
+  children: [
+    {
+      id: '1',
+      children: [
+        { id: '1a', children: [{ id: '1ai', children: [] }] },
+        { id: '1b', children: [] },
+        {
+          id: '1c',
+          children: [{ id: '1ci', children: [] }],
+        },
+      ],
+    },
+    {
+      id: '2',
+      children: [
+        { id: '2a', children: [] },
+        {
+          id: '2b',
+          children: [
+            { id: '2bi', children: [] },
+            { id: '2bii', children: [] },
+          ],
+        },
+      ],
+    },
+    {
+      id: '3',
+      children: [],
+    },
+  ],
+};
+
+describe('threads', function() {
+  let fakeStore;
+  let service;
+
+  beforeEach(() => {
+    fakeStore = {
+      setForceVisible: sinon.stub(),
+    };
+    service = threadsService(fakeStore);
+  });
+
+  describe('#forceVisible', () => {
+    [
+      'top',
+      '1',
+      '2',
+      '3',
+      '1a',
+      '1b',
+      '1c',
+      '2a',
+      '2b',
+      '1ai',
+      '1ci',
+      '2bi',
+      '2bii',
+    ].forEach(threadId => {
+      it('should set the thread and its children force-visible in the store', () => {
+        service.forceVisible(NESTED_THREADS);
+        assert.calledWith(fakeStore.setForceVisible, threadId);
+      });
+    });
+
+    it('should not set the visibility on thread ancestors', () => {
+      // This starts at the level with `id` of '1'
+      service.forceVisible(NESTED_THREADS.children[0]);
+
+      const calledWithThreadIds = [];
+      for (let i = 0; i < fakeStore.setForceVisible.callCount; i++) {
```

Comment:
These gymnastics because we use `assert` which doesn't give us a negation construct (e.g. cf. `expect.not`)—I'm trying to test for the inverse of `assert.calledWith` (i.e. equivalent to "not called with").

So we're looking at which threads it _was_ called on and making sure none of them are the ancestors, which are the true subject of the test here.